internal/venice-consumer/src/main/java/com/linkedin/venice/client/consumer/VeniceChangelogConsumer.java

```diff
@@ -1,22 +1,24 @@
 package com.linkedin.venice.client.consumer;
 
-import java.util.Map;
+import com.linkedin.venice.annotation.Experimental;
 import java.util.Set;
 import java.util.concurrent.CompletableFuture;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
```

Comment:
Let us not leak a Kafka API as part of our interface. Let's return our own types only.

Reply:
Just want to confirm, did you mean by using `VeniceConsumerRecords<K, V>`?

Reply:
Did you mean by just using existing interfaces like `VeniceConsumerRecords<K, V>` instead of a map-like Kafka API `ConsumerRecords<K,V>`?

Reply:
Yeah, that was my intention as well.  Don't use Kafka's ConsumerREcord, lets just make our own.

Reply:
That is why I proposed the other set of interfaces to have our own pub sub abstraction, the other day (still need to convert it into a proper PR and republish). Regarding whether we wish to return just one record at a time or a collection of records, either way is fine I guess.

Reply:
I think a collection of records might be fine. How about I put `VeniceConsumerRecords<K, V>` here and update `VeniceConsumerRecords<K, V>` a bit to incorporate your pub/sub changes after you merge it?

Reply:
Or maybe incorporate the pub sub interfaces from my change into your change. I didn't mean to hold progress back (and sorry that my availability has been spotty lately). I just wanted to give an example of how it might be done, but you can take it over if it looks acceptable to you.

Reply:
Sure, I can give it a try.